zipkin-lens/src/components/GlobalSearch/GlobalDropdownMenu.js

```diff
@@ -0,0 +1,139 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withRouter } from 'react-router';
+import Modal from 'react-modal';
+
+const propTypes = {
+  history: PropTypes.shape({
+    push: PropTypes.func.isRequired,
+  }).isRequired,
+  setTrace: PropTypes.func.isRequired,
+};
+
+// This selector (class name) is used to specify a modal parent component.
+const modalWrapperClass = 'global-dropdown-menu__modal-wrapper';
+
+class GlobalDropdownMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.fileInpueElement = undefined;
+    this.state = {
+      isModalOpened: false,
+      traceId: '',
+    };
+    this.handleOpenModalToggle = this.handleOpenModalToggle.bind(this);
+    this.handleTraceIdButtonClick = this.handleTraceIdButtonClick.bind(this);
+    this.handleTraceIdChange = this.handleTraceIdChange.bind(this);
+    this.handleTraceJsonChange = this.handleTraceJsonChange.bind(this);
+  }
+
+  handleOpenModalToggle() {
+    const { isModalOpened } = this.state;
+    this.setState({ isModalOpened: !isModalOpened });
+  }
+
+  handleTraceIdButtonClick(event) {
+    const { history } = this.props;
+    const { traceId } = this.state;
+    history.push({
+      pathname: `/zipkin/traces/${traceId}`,
+    });
+    this.setState({ isModalOpened: false });
+    event.stopPropagation();
+  }
+
+  handleTraceIdChange(event) {
+    this.setState({
+      traceId: event.target.value,
+    });
+  }
+
+  handleTraceJsonChange(event) {
+    const { history, setTrace } = this.props;
+
+    const [file] = event.target.files;
+    const fileReader = new FileReader();
+
+    fileReader.onload = () => {
+      const { result } = fileReader;
+      let rawTrace;
+      try {
+        rawTrace = JSON.parse(result);
+        setTrace(rawTrace);
+        history.push({
+          pathname: '/zipkin/traceViewer',
+        });
+      } catch (error) {
+        // Do nothing
```

Comment:
maybe alert the json is malformed? I assume this catch will result in someone uploading an incorrect file (ex just one span) and getting no feedback about it. correct?

Reply:
Yes.
Do you think that it is better to display the alert dialog?

Reply:
old UI did something like this..

![Screen Shot 2019-03-19 at 9 25 51 AM](https://user-images.githubusercontent.com/64215/54573874-0b95ec00-4a29-11e9-9d9f-662b866241a4.png)

Reply:
OK, Thank you.
I'll do so